src/algorithms/tracking/TrackerSourceLinker.cc

```diff
@@ -80,19 +81,24 @@ eicrecon::TrackerSourceLinkerResult *eicrecon::TrackerSourceLinker::produce(std:
 
         Acts::Vector2 loc = Acts::Vector2::Zero();
         Acts::Vector2 pos;
+        auto hit_det = hit->getCellID()&0xFF;
+        auto onSurfaceTolerance = 0.1*Acts::UnitConstants::um;      // By default, ACTS uses 0.1 micron as the on surface tolerance
+        if (hit_det==dd4hepGeo->constant<int>("B0Tracker_Station_1_ID")){
```

Comment:
Now I'd just move this lookup into initialization instead of doing this explicitly again on every hit.

Reply:
I meant putting in the `init` method.

Reply:
So, declare a variable for m_dd4hepGeo and m_b0detid? Something like that?

Reply:
Yeah, for example. I'm worried that the lookup of the constant is expensive since it's a string comparison over all possible constants in the geometry, which are probably not stored in a structure that is optimized for these lookups (and maybe not even a single structure but distributed across detector elements).

Reply:
Please check now